src/components/ClaimENSPage/ClaimENSPage.tsx

```diff
@@ -53,20 +104,25 @@ export default class ClaimENSPage extends React.PureComponent<Props, State> {
                 </span>
               </Section>
               <Form onSubmit={this.handleClaim}>
-                <Field
-                  label="Name"
-                  value={name}
-                  message="Names cannot contain non-alphanumeric characters or spaces."
-                  action={`${name.length}/${MAX_NAME_SIZE}`}
-                  error={name.length > 1 && !isValid}
-                  onChange={this.handleNameChange}
-                  onAction={this.handleOnAction}
-                />
+                <Section>
+                  <Field
+                    label="Name"
+                    value={name}
+                    message="Names cannot contain non-alphanumeric characters or spaces."
+                    action={`${name.length}/${MAX_NAME_SIZE}`}
+                    error={name.length > 1 && !isValid}
+                    onChange={this.handleNameChange}
+                    onAction={this.handleOnAction}
+                  />
+                </Section>
+                <Section>
+                  <Radio toggle checked={this.isManaApproved()} onChange={this.handleManaApprove} label={'MANA approved'} />
+                </Section>
                 <Row className="actions">
                   <Button className="cancel" onClick={onBack}>
                     {t('global.cancel')}
                   </Button>
-                  <Button type="submit" primary disabled={!isValid}>
+                  <Button type="submit" primary disabled={!isValid || !this.isManaApproved()} loading={this.state.isLoading}>
```

Comment:
You're already extracting the name from the state, you can add isLoading there right? (`L83`)